switchover/dbsync/shell.go

```diff
@@ -294,12 +356,19 @@ func RunShell(oldURL, newURL string) error {
 				return nil
 			}
 
-			start := time.Now()
-			err = s.Sync(ctx, false, false)
-			if err != nil {
-				return err
+			for {
+				start := time.Now()
+				err = s.Sync(ctx, false, false)
+				if err != nil {
+					return err
+				}
+				dur := time.Since(start).Truncate(time.Second / 10)
+				sh.Printf("Completed sync in %s\n", dur.String())
+				if dur < maxSync {
+					break
+				}
+				sh.Println("Took longer than max sync time, re-syncing")
```

Comment:
potential to become an infinite loop here?

Reply:
Yes, intentionally. It will keep re-syncing the DB until it catches up, or the user uses CTRL+C to stop the process.

It's hard to predict how many times it may take, and if it were to stop at an arbitrary number, a busy system could fall behind again.

Since it's safe to run multiple times and can be interrupted at any time, it will keep going as long as is needed this way, rather than having the user re-run `execute` and confirm over and over.